slider/collection.py

```diff
@@ -0,0 +1,98 @@
+# TODO: not use replay private functions
+from .replay import _consume_int, _consume_string
+
+
+class CollectionDB:
+    """An osu! ``collection.db`` file.
+
+    Parameters
+    ----------
+    version : int
+        Version number (e.g. 20150203)
+    num_collections : int
+        Number of collections
+    collections : list[Collection]
+        List of :class:`~slider.collection.Collection` s
+    """
+    def __init__(self, version, num_collections, collections):
+        self.version = version
+        self.num_collections = num_collections
+        self.collections = collections
+
+    @classmethod
+    def from_path(cls, path):
+        """Read in a ``collection.db`` file from disk.
+
+        Parameters
+        ----------
+        path : str or pathlib.Path
+            The path to the file to read from.
+        """
+        with open(path, 'rb') as f:
+            return cls.from_file(f)
+
+    @classmethod
+    def from_file(cls, file):
+        """Read in a ``collection.db`` file from an open file object.
+
+        Parameters
+        ----------
+        file : file-like
+            The file object to read from.
+        """
+        return cls.parse(file.read())
+
+    @classmethod
+    def parse(cls, data):
+        """Parse from ``collection.db`` data.
+
+        Parameters
+        ----------
+        data : bytes
+            The data from a ``collection.db`` file.
+        """
+        buffer = bytearray(data)
+
+        version = _consume_int(buffer)
+        num_collections = _consume_int(buffer)
+        collections = []
+        for i in range(num_collections):
+            collections.append(Collection.parse(buffer))
+
+        return cls(version, num_collections, collections)
+
+
+class Collection:
```

Comment:
How do you plan to use this? Should we add a method which takes a `Library` and returns the constituent beatmaps?

Reply:
If you mean `Collection` class, currently it exists only to help `CollectionDB`. It is not intended to be created by the user. 
Eventually I want to add `Library` support so that we have the actual beatmap associated with MD5. I need to investigate how `Library` works first before adding this in. I think it's ok to leave it out for now and get the basic functionality in.